File: java/src/main/java/it/giannibombelli/wsc2026/booking/domain/booking/BookingStatus.java
package it.giannibombelli.wsc2026.booking.domain.booking;

public enum BookingStatus {
    PLACED("placed"),
    CONFIRMED("confirmed"),
    REFUSED("refused"),
    REJECTED("rejected");

    private final String value;

    BookingStatus(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: typescript/src/booking/domain/booking/bookingStatus.ts
// Stati del ciclo di vita di una prenotazione.

export const BookingStatus = {
  PLACED: 'PLACED',
  CONFIRMED: 'CONFIRMED',
  REFUSED: 'REFUSED',
  REJECTED: 'REJECTED',
} as const;

export type BookingStatus = (typeof BookingStatus)[keyof typeof BookingStatus];
